src/io/iohk/scalanet/peergroup/UDPPeerGroup.scala

```
@@ -52,20 +62,30 @@ class UDPPeerGroup[F[_]](val udpPeerGroupConfig: Config)(implicit liftF: Lift[F]
   }
 
   private def writeUdp(address: InetSocketAddress, data: ByteBuffer): Unit = {
-    val udp = DatagramChannel.open()
-    udp.configureBlocking(true)
-    udp.connect(address)
     try {
-      udp.write(data)
-    } finally {
-      udp.close()
+      val udp = DatagramChannel.open()
+      udp.configureBlocking(true)
+      udp.connect(address)
+      try {
+        udp.write(data)
+      } finally {
+        udp.close()
+      }
+    } catch {
+      case e: Throwable => e.printStackTrace()
```

Don't we need to propagate the error?

---

Yes, we do. Thanks.